### PR TITLE
Changes to links & filenames, little bit of SEO

### PR DESCRIPTION
**Pull request by jyooru**
## TL;DR: tidy up file names and links
- **`_` in file and folder names have been changed to `-`**. I did this for SEO.
  >Google treats a hyphen as a word separator, but does not treat an underscore that way. Google treats and underscore as a word joiner — so red_sneakers is the same as redsneakers to Google. This has been confirmed directly by Google themselves, including the fact that using dashes over underscores will have a (minor) ranking benefit.
https://www.ecreativeim.com/blog/index.php/2011/03/30/seo-basics-hyphen-or-underscore-for-seo-urls/
- **Above, but also for links inside files**
-  **Relative links** - `https://wiki.bedrock.dev/` has been replaced with `/` throughout files. So `https://wiki.bedrock.dev/some/page` has been swapped with `/some/page`. This is better because if you move to a different website all your links would be pointing to the old website, which doesn't exist anymore.
- With these changes, I made a **little script for 404 pages** (`/assets/js/404.js`) *(my first JS script!)* that just in case there is an old link, it will:
  1. Make all letters lowercase
  1. Change all `_` to `-`
  ###### Feel free to modify this one ↑, it is my first script in JavaScript.
- **Removed `.html` from links**
  Really does nothing, just looks a little cleaner in the address bar 😎
  I couldn't figure out how to do it for [getting started](https://wiki.bedrock.dev/getting_started.html) (only one that needs changing) tho

If there are any changes that need to be made I'll be happy to help too!
Lmk if anything above needs changing too, I'm in the bedrock wiki discord.

### CODE REVIEW

**Comment by SirLich** [comment]
This looks awesome! Thanks for the PR!

Did you run it locally in Jekyll?

**Comment by jyooru** [comment]
> Did you run it locally in Jekyll?

Not yet, my PC doesn't like installing Jekyll and I have to install it through windows subsystem for linux. I'll do that today.

**Comment by jyooru** [comment]
Alright, Jekyll installed and found one bug, fixed now. Everything else seems to work.

> * **Removed `.html` from links**
>   Really does nothing, just looks a little cleaner in the address bar 😎
>   I couldn't figure out how to do it for [getting started](https://wiki.bedrock.dev/getting_started.html) (only one that needs changing) tho

I did find that most links in the navigation bar are pointing to `.html`, I'm not sure how to fix it but again it is only shows in the navigation bar.

**Comment by SirLich** [comment]
Hi! Thanks again for the PR!

Due to some updates in a few tutorial docs, there are now some merge conflicts. 

Would you mind merging in the most recent gh-pages branch, and dealing with the merge conflicts, then pushing up again? Then this will be accepted.

**Comment by jyooru** [comment]
Okay, I've merged the conflicts.

I noticed that **tutorials/materials** should be **tutorials/materials.md**, but I haven't changed it.
